source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -353,16 +353,21 @@ void ShaderNode::initialize(const Node& node, const NodeDef& nodeDef, GenContext
                 }
             }
             const string& valueString = portValue ? portValue->getValueString() : EMPTY_STRING;
-            std::pair<TypeDesc, ValuePtr> enumResult;
-            const string& enumNames = nodeDefInput->getAttribute(ValueElement::ENUM_ATTRIBUTE);
-            const TypeDesc type = context.getTypeDesc(nodeDefInput->getType());
-            if (context.getShaderGenerator().getSyntax().remapEnumeration(valueString, type, enumNames, enumResult))
```

Comment:
The "bug" appears to be in GLSL and MSL `remapEnumeration` method. It is returning true when there is no value mapped. MDL returns the correct value of `false` in this case.